store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * ***** BEGIN LICENSE BLOCK *****
+ * Zimbra Collaboration Suite Server
+ * Copyright (C) 2022 Synacor, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software Foundation,
+ * version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ * ***** END LICENSE BLOCK *****
+ */
+
+package com.zimbra.cs.service.util;
+
+import org.json.JSONException;
+
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.util.StringUtil;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.volume.Volume;
+import com.zimbra.cs.volume.VolumeServiceException;
+import com.zimbra.soap.admin.message.CreateVolumeRequest;
+import com.zimbra.soap.admin.type.VolumeInfo;
+import com.zimbra.util.ExternalVolumeInfoHandler;
+
+public class VolumeConfigUtil {
+
+    /**
+     * Validate the create volume request parameters
+     *
+     * @param request
+     * @return
+     * @throws ServiceException
+     */
+    public static void validateCreateVolumeRequest(CreateVolumeRequest request, VolumeInfo volInfoRequest,
+                                                   Volume.StoreType enumStoreType) throws ServiceException {
+        ExternalVolumeInfoHandler extVolInfoHandler = new ExternalVolumeInfoHandler(Provisioning.getInstance());
+
+        // validate store type
+        Short storeType = volInfoRequest.getStoreType();
+        if (1 != storeType.intValue() && 2 != storeType.intValue()) {
```

Comment:
I think this statement should look like.
`if (StoreType.INTERNAL.getStoreType() != storeType.intValue() && StoreType.EXTERNAL.getStoreType()  != storeType.intValue())`

Reply:
Done